modules/utxo-bin/src/hollowSegwitSpend.ts

```diff
@@ -0,0 +1,62 @@
+import * as utxolib from '@bitgo/utxo-lib';
+
+/**
+ * Users can mistakenly create wrapped segwit outputs on chains that do not support it (bitcoincash, bitcoinSV).
+ *
+ * These funds can be claimed by miners that learn the hash preimage, typically by observing a spend on the bitcoin
+ * chain.
+ *
+ * Since the transaction does not have a full inner script, let's call them "hollow".
+ *
+ * See https://bitcoin.stackexchange.com/a/115850/137601 for more details.
+ */
+export type HollowSegwitSpend =
+  /** Wrapped Segwit Pay-To-Witness-Script-Hash */
+  | { scriptType: 'p2shP2wshHollow' }
+  /** Wrapped Segwit Pay-To-Pubkey-Hash */
+  | { scriptType: 'p2shP2wpkhHollow' };
+
+/**
+ * @return HollowSegwitSpend if spend is a
+ */
+export function parseHollowSegwitSpend(
+  input: utxolib.TxInput,
+  network: utxolib.Network
+): HollowSegwitSpend | undefined {
+  if (
+    utxolib.getMainnet(network) !== utxolib.networks.bitcoincash &&
+    utxolib.getMainnet(network) !== utxolib.networks.bitcoinsv
+  ) {
+    // this can only happen on the BCH/BSV forks
```

Comment:
BSV even did away with P2SH I believe, so there can't be new P2SH outputs created. Although I suppose it's possible that there are old "hollow" outputs out there.

Reply:
right I remember now